moveit_core/planning_interface/include/moveit/planning_interface/planning_interface.h

```diff
@@ -117,6 +120,9 @@ class PlanningContext
   /** \brief Set the planning request for this context */
   void setMotionPlanRequest(const MotionPlanRequest& request);
 
+  /** \brief Set the planning request for this context */
+  void setStateCostFunction(const StateCostFn& state_cost_function);
```

Comment:
I think that state cost functions should be provided with the planning request, just as constraints